### PR TITLE
Remove dieharder tests

### PR DESCRIPTION
**Pull request by amirhosv**
*Description of changes:*
ACCP relies on AWS-LC for SecureRandom implementation and we rely on AWS-LC for the testing this component. The flaky nature of these tests is the reason for us to remove them.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/ci/README.md [hunk 1]
```diff
@@ -52,13 +52,12 @@ CodeBuild|gcc 7|corretto 17|aarch|Ubuntu 20.04|FIPS/non-FIPS
 (macOS CI dimension is currently disabled, go to the Actions tab in the main repo to enable it when its ready.)
 
 
-### Dieharder & Overkill tests
+### Overkill tests
```
Comment (WillChilds-Klein) [inline tests/ci/README.md:55]: looks like we've also removed the overkill tests?

File: tests/ci/codebuild/release/al2_aarch64_test.yml [hunk 2]
```diff
@@ -1,54 +0,0 @@
-# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-# SPDX-License-Identifier: Apache-2.0
-
-version: 0.2
-
-env:
-  variables:
```
Comment (WillChilds-Klein) [inline tests/ci/codebuild/release/al2_aarch64_test.yml:7]: how is the removal of this and other codebuild specs related to removing dieharder tests?

File: tests/ci/README.md (hunk 1 above)
Comment (amirhosv) [reply tests/ci/README.md:55]: [The overkill tests](https://github.com/corretto/amazon-corretto-crypto-provider/blob/main/build.gradle#L568) task has been modified by removing dieharder from it. It is still there.

File: tests/ci/codebuild/release/al2_aarch64_test.yml (hunk 2 above)
Comment (amirhosv) [reply tests/ci/codebuild/release/al2_aarch64_test.yml:7]: Not related. It was dead code, so I just removed those too. This is me being lazy and not creating two PRs. I can bring them back and submit another PR for it.

**Review by geedo0 — APPROVED** [review]

File: tests/ci/codebuild/release/al2_aarch64_test.yml (hunk 2 above)
Comment (WillChilds-Klein) [reply tests/ci/codebuild/release/al2_aarch64_test.yml]: nah, no worries.

**Review by WillChilds-Klein — APPROVED** [review]

**Pull request merged**